<?xml version="1.0" encoding="UTF-8"?><diffreport><diff><table border="0" cellpadding="1" cellspacing="0" width="100%"><tr><td><h3 class="ctxid">class:ConcurrentModificationException [NONE]</h3></td></tr><tr><td><table border="0" cellpadding="5" cellspacing="1" bgcolor="black" width="100%"><tr valign="top"><td bgcolor="#FFFFFF" colspan="1"><div class="description"> <a start_ctx_id="class:ConcurrentModificationException"/><ul class="blockList"><li class="blockList"><dl> <dt>All Implemented Interfaces:  </dt><dd> <a href="../../java/io/Serializable.html" title="interface in java.io">Serializable</a> </dd> </dl><dl> <dt>Direct Known Subclasses:  </dt><dd> <a href="../../java/nio/file/DirectoryIteratorException.html" title="class in java.nio.file">DirectoryIteratorException</a> </dd> </dl> <hr/> <br/><pre>public class <span class="typeNameLabel">ConcurrentModificationException</span>
extends <a href="../../java/lang/RuntimeException.html" title="class in java.lang">RuntimeException</a></pre><div class="block">This exception may be thrown by methods that have detected concurrent modification of an object when such modification is not permissible.<p> For example, it is not generally permissible for one thread to modify a Collection while another thread is iterating over it. In general, the results of the iteration are undefined under these circumstances. Some Iterator implementations (including those of all the general purpose collection implementations provided by the JRE) may choose to throw this exception if this behavior is detected. Iterators that do this are known as <i>fail-fast</i> iterators, as they fail quickly and cleanly, rather that risking arbitrary, non-deterministic behavior at an undetermined time in the future.                                                                                        </p><p> Note that this exception does not always indicate that an object has been concurrently modified by a <i>different</i> thread. If a single thread issues a sequence of method invocations that violates the contract of an object, the object may throw this exception. For example, if a thread modifies a collection directly while it is iterating over the collection with a fail-fast iterator, the iterator will throw this exception.                                                                    </p><p>Note that fail-fast behavior cannot be guaranteed as it is, generally speaking, impossible to make any hard guarantees in the presence of unsynchronized concurrent modification. Fail-fast operations throw <code>ConcurrentModificationException</code> on a best-effort basis. Therefore, it would be wrong to write a program that depended on this exception for its correctness: <i><code>ConcurrentModificationException</code> should be used only to detect bugs.       </i>                                                  </p>                    </div><dl> <dt> <span class="simpleTagLabel">Since:</span>   </dt><dd>1.2</dd> <dt> <span class="seeLabel">See Also: </span>   </dt><dd> <a href="../../java/util/Collection.html" title="interface in java.util"><code>Collection</code></a>, <a href="../../java/util/Iterator.html" title="interface in java.util"><code>Iterator</code></a>, <a href="../../java/util/Spliterator.html" title="interface in java.util"><code>Spliterator</code></a>, <a href="../../java/util/ListIterator.html" title="interface in java.util"><code>ListIterator</code></a>, <a href="../../java/util/Vector.html" title="class in java.util"><code>Vector</code></a>, <a href="../../java/util/LinkedList.html" title="class in java.util"><code>LinkedList</code></a>, <a href="../../java/util/HashSet.html" title="class in java.util"><code>HashSet</code></a>, <a href="../../java/util/Hashtable.html" title="class in java.util"><code>Hashtable</code></a>, <a href="../../java/util/TreeMap.html" title="class in java.util"><code>TreeMap</code></a>, <a href="../../java/util/AbstractList.html" title="class in java.util"><code>AbstractList</code></a>, <a href="../../serialized-form.html#java.util.ConcurrentModificationException">Serialized Form </a>           </dd>  </dl>  </li></ul> </div> <a end_ctx_id="class:ConcurrentModificationException"/> </td></tr></table></td></tr></table><table border="0" cellpadding="1" cellspacing="0" width="100%"><tr><td><h3 class="ctxid">constructor:ConcurrentModificationException-- [NONE]</h3></td></tr><tr><td><table border="0" cellpadding="5" cellspacing="1" bgcolor="black" width="100%"><tr valign="top"><td bgcolor="#FFFFFF" colspan="1"> <a start_ctx_id="constructor:ConcurrentModificationException--"/><a name="ConcurrentModificationException--"/><ul class="blockList"><li class="blockList"><h4>ConcurrentModificationException</h4><pre>public ConcurrentModificationException()</pre><div class="block">Constructs a ConcurrentModificationException with no detail message.      </div></li></ul> <a end_ctx_id="constructor:ConcurrentModificationException--"/>  </td></tr></table></td></tr></table><table border="0" cellpadding="1" cellspacing="0" width="100%"><tr><td><h3 class="ctxid">constructor:ConcurrentModificationException-java.lang.String- [NONE]</h3></td></tr><tr><td><table border="0" cellpadding="5" cellspacing="1" bgcolor="black" width="100%"><tr valign="top"><td bgcolor="#FFFFFF" colspan="1"> <a start_ctx_id="constructor:ConcurrentModificationException-java.lang.String-"/><a name="ConcurrentModificationException-java.lang.String-"/><ul class="blockList"><li class="blockList"><h4>ConcurrentModificationException</h4><pre>public ConcurrentModificationException(<a href="../../java/lang/String.html" title="class in java.lang">String</a> message)</pre><div class="block">Constructs a <code>ConcurrentModificationException</code> with the specified detail message.       </div><dl> <dt> <span class="paramLabel">Parameters:</span>   </dt><dd> <code>message</code> - the detail message pertaining to this exception.         </dd> </dl></li></ul> <a end_ctx_id="constructor:ConcurrentModificationException-java.lang.String-"/>  </td></tr></table></td></tr></table><table border="0" cellpadding="1" cellspacing="0" width="100%"><tr><td><h3 class="ctxid">constructor:ConcurrentModificationException-java.lang.Throwable- [NONE]</h3></td></tr><tr><td><table border="0" cellpadding="5" cellspacing="1" bgcolor="black" width="100%"><tr valign="top"><td bgcolor="#FFFFFF" colspan="1"> <a start_ctx_id="constructor:ConcurrentModificationException-java.lang.Throwable-"/><a name="ConcurrentModificationException-java.lang.Throwable-"/><ul class="blockList"><li class="blockList"><h4>ConcurrentModificationException</h4><pre>public ConcurrentModificationException(<a href="../../java/lang/Throwable.html" title="class in java.lang">Throwable</a> cause)</pre><div class="block">Constructs a new exception with the specified cause and a detail message of <code>(cause==null ? null : cause.toString())    </code> (which typically contains the class and detail message of <code>cause</code>.                       </div><dl> <dt> <span class="paramLabel">Parameters:</span>   </dt><dd> <code>cause</code> - the cause (which is saved for later retrieval by the <a href="../../java/lang/Throwable.html#getCause--"><code>Throwable.getCause()</code></a> method). (A <code>null</code> value is permitted, and indicates that the cause is nonexistent or unknown.)                            </dd> <dt> <span class="simpleTagLabel">Since:</span>   </dt><dd>1.7</dd>  </dl></li></ul> <a end_ctx_id="constructor:ConcurrentModificationException-java.lang.Throwable-"/>  </td></tr></table></td></tr></table><table border="0" cellpadding="1" cellspacing="0" width="100%"><tr><td><h3 class="ctxid">constructor:ConcurrentModificationException-java.lang.String-java.lang.Throwable- [NONE]</h3></td></tr><tr><td><table border="0" cellpadding="5" cellspacing="1" bgcolor="black" width="100%"><tr valign="top"><td bgcolor="#FFFFFF" colspan="1"> <a start_ctx_id="constructor:ConcurrentModificationException-java.lang.String-java.lang.Throwable-"/><a name="ConcurrentModificationException-java.lang.String-java.lang.Throwable-"/><ul class="blockListLast"><li class="blockList"><h4>ConcurrentModificationException</h4><pre>public ConcurrentModificationException(<a href="../../java/lang/String.html" title="class in java.lang">String</a> message,
                                       <a href="../../java/lang/Throwable.html" title="class in java.lang">Throwable</a> cause)</pre><div class="block">Constructs a new exception with the specified detail message and cause.<p>Note that the detail message associated with <code>cause</code> is <i>not</i> automatically incorporated in this exception's detail message.                </p>          </div><dl> <dt> <span class="paramLabel">Parameters:</span>   </dt><dd> <code>message</code> - the detail message (which is saved for later retrieval by the <a href="../../java/lang/Throwable.html#getMessage--"><code>Throwable.getMessage()</code></a> method).               </dd><dd> <code>cause</code> - the cause (which is saved for later retrieval by the <a href="../../java/lang/Throwable.html#getCause--"><code>Throwable.getCause()</code></a> method). (A <code>null</code> value is permitted, and indicates that the cause is nonexistent or unknown.)                            </dd> <dt> <span class="simpleTagLabel">Since:</span>   </dt><dd>1.7</dd>  </dl></li></ul> <a end_ctx_id="constructor:ConcurrentModificationException-java.lang.String-java.lang.Throwable-"/>  </td></tr></table></td></tr></table></diff></diffreport>